pkg/aws/actuator/actuator.go

```diff
@@ -263,11 +264,82 @@ func (a *AWSActuator) getProviderSpec(cr *minterv1.CredentialsRequest) (*minterv
 	return nil, fmt.Errorf("no providerSpec defined")
 }
 
-// sync handles both create and update idempotently.
 func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) error {
+
 	logger := a.getLogger(cr)
 	logger.Debug("running sync")
 
+	// Should we update anything
+	needsUpdate, err := a.needsUpdate(ctx, cr)
+	if err != nil {
+		logger.WithError(err).Error("error determining whether a credentials update is needed")
+		return &actuatoriface.ActuatorError{
+			ErrReason: minterv1.CredentialsProvisionFailure,
+			Message:   fmt.Sprintf("error determining whether a credentials update is needed: %v", err),
+		}
+	}
+
+	if !needsUpdate {
+		return nil
+	}
+
+	cloudCredsSecret, err := a.getCloudCredentialsSecret(ctx, logger)
+	if err != nil {
+		logger.WithError(err).Error("issue with cloud credentials secret")
+		return err
+	}
+
+	if cloudCredsSecret.Annotations[secretannotator.AnnotationKey] == secretannotator.InsufficientAnnotation {
+		msg := "cloud credentials insufficient to satisfy credentials request"
+		logger.Error(msg)
+		return &actuatoriface.ActuatorError{
+			ErrReason: minterv1.InsufficientCloudCredentials,
+			Message:   msg,
+		}
+	}
+
+	if cloudCredsSecret.Annotations[secretannotator.AnnotationKey] == secretannotator.PassthroughAnnotation {
```

Comment:
With these big forks in logic it might be worth doing a credPassthroughStrategy and credMintStrategy, check the annotation early in the actuator, create the right implementation, and then move any specific sync logic and needsupdate logic to the strategies.